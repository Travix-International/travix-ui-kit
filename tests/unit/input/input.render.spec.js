import { shallow } from 'enzyme';
import React from 'react';
import Input from '../../../components/input/input';

describe('Input', () => {
  describe('#render()', () => {
    const onChange = () => {};

    it('should render correct input', () => {
      const wrapper = shallow(
        <Input
          disabled
          name="input"
          onChange={onChange}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should not modify mods', () => {
      const mods = ['test'];
      const wrapper = shallow(
        <Input
          mods={mods}
        />
      );

      expect(mods.length).toEqual(1);
      expect(mods[0]).toEqual('test');
      expect(wrapper).toMatchSnapshot();
    });

    it('should render multiline input', () => {
      const wrapper = shallow(
        <Input
          disabled
          multiline
          name="input"
          onChange={onChange}
        />
      );

      expect(wrapper).toMatchSnapshot();
    });

    it('should render span for icon', () => {
      const wrapper = shallow(
        <Input
          name="input"
          onChange={onChange}
          status="error"
        />
      );

      expect(wrapper).toMatchSnapshot();
    });
  });
});
