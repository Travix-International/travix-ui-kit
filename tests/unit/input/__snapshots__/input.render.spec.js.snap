// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input #render() should not modify mods 1`] = `
<div
  className="ui-input-container"
>
  <input
    className="ui-input ui-input_test"
    disabled={false}
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
</div>
`;

exports[`Input #render() should render correct input 1`] = `
<div
  className="ui-input-container"
>
  <input
    className="ui-input ui-input_disabled"
    disabled={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
</div>
`;

exports[`Input #render() should render multiline input 1`] = `
<div
  className="ui-input-container"
>
  <textarea
    className="ui-input ui-input_disabled"
    disabled={true}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
</div>
`;

exports[`Input #render() should render span for icon 1`] = `
<div
  className="ui-input-container"
>
  <input
    className="ui-input ui-input_error"
    disabled={false}
    name="input"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
  />
  <span
    className="ui-input-status"
  />
</div>
`;
